### PR TITLE
fix 'The document's methods were overridden incorrectly' (close #741)

### LINKED ISSUE #741: The document's methods were overridden incorrectly
```javascript
var htmlDoc = document.implementation.createHTMLDocument("title");
var body = htmlDoc.getElementsByTagName("body")[0];
```
The `getElementsByTagName` function returns the body of the `document` instead of the body of `htmlDoc`.

### PR DESCRIPTION
**Pull request by LavrovArtem**
/cc @churkin @georgiy-abbasov

### CODE REVIEW

File: test/client/fixtures/sandbox/shadow-ui-test.js [hunk 1]
```diff
@@ -628,3 +649,11 @@ asyncTest('isShadowContainerCollection for cross-domain iframe.contentWindow mus
         });
     document.body.appendChild(crossDomainIframe);
 });
+
+if (document.implementation && document.implementation.createHTMLDocument) {
+    test('the getElementsByTagName function must return the body of htmlDoc (GH-741)', function () {
+        var htmlDoc = document.implementation.createHTMLDocument('title');
+
+        ok(htmlDoc.getElementsByTagName('body')[0] === nativeMethods.getElementsByTagName.call(htmlDoc, 'body')[0]);
```
Comment (churkin) [inline test/client/fixtures/sandbox/shadow-ui-test.js]: `strictEqual`

**Comment by LavrovArtem** [comment]
FPR

**Comment by georgiy-abbasov** [comment]
lgtm

**Comment by LavrovArtem** [comment]
ping @churkin

**Comment by churkin** [comment]
lgtm

**Pull request merged**